src/pages/permissions/panels/PermissionSelector.tsx

```diff
@@ -0,0 +1,159 @@
+import { Button, Select, useNotify } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { fetchPermissions } from "api/auth-permissions";
+import { ChangeEvent, FC, useEffect } from "react";
+import {
+  defaultOption,
+  generateResourceAndEntitlementOptions,
+  noneAvailableOption,
+  resourceTypeOptions,
+} from "util/permissions";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  resourceType: string;
+  resource: string;
+  entitlement: string;
+  onChangeResourceType: (resourceType: string) => void;
+  onChangeResource: (resource: string) => void;
+  onChangeEntitlement: (entitlement: string) => void;
+  onAddPermission: () => void;
+}
+
+const PermissionSelector: FC<Props> = ({
+  resourceType,
+  resource,
+  entitlement,
+  onChangeResourceType,
+  onChangeResource,
+  onChangeEntitlement,
+  onAddPermission,
+}) => {
+  const notify = useNotify();
+
+  const {
+    data: permissions,
+    isLoading: isPermissionsLoading,
+    error: permissionsError,
+  } = useQuery({
+    queryKey: [queryKeys.permissions, resourceType],
+    queryFn: () => fetchPermissions({ resourceType }),
+    enabled: !!resourceType,
+  });
+
+  useEffect(() => {
+    document.getElementById("resourceType")?.focus();
+  }, []);
+
+  useEffect(() => {
+    if (!resourceType) {
+      document.getElementById("resourceType")?.focus();
+      return;
+    }
+
+    if (resourceType === "server") {
+      document.getElementById("entitlement")?.focus();
+      return;
+    }
+    document.getElementById("resource")?.focus();
+  }, [resourceType, permissions]);
+
+  useEffect(() => {
+    document.getElementById("entitlement")?.focus();
+  }, [resource]);
+
+  useEffect(() => {
+    document.getElementById("add-entitlement")?.focus();
+  }, [entitlement]);
+
+  const handleResourceTypeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    const resourceType = e.target.value;
+    onChangeEntitlement("");
+    onChangeResourceType(resourceType);
+
+    // for server resource type we can select a resource, so automatically set it
+    // label for resource will show "server" if resource type is set to server
+    if (resourceType === "server") {
+      onChangeResource("/1.0");
+    } else {
+      onChangeResource("");
+    }
+  };
+
+  const handleResourceChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    onChangeResource(e.target.value);
+  };
+
+  const handleEntitlementChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    onChangeEntitlement(e.target.value);
+  };
+
+  if (permissionsError) {
+    notify.failure("Loading permissions failed", permissionsError);
+  }
+
+  const { resourceOptions, resourceEntitlementOptions } =
+    generateResourceAndEntitlementOptions(resourceType, permissions);
```

Comment:
Internally this is missing a call to `decodeURIComponent`, as some resources can contain spaces and we should show them as such instead of as `%20`.

This is relevant for the dropdown on adding permissions, but also for the display of existing permissions in the section below.

![image](https://github.com/canonical/lxd-ui/assets/1155472/e75f2ead-7ae6-4206-ba8e-9a70df909182)

Reply:
Apart from encoding, some entities are better resolved. For images and identities, we should probably display a more recognizable name than just the ID (see screen below).

Should we remove the resource type "Certificate" from the dropdown? It seems to be already included in the resource type "Identity".

![image](https://github.com/canonical/lxd-ui/assets/1155472/ae2068a6-a92a-4517-934c-ac53d5860cf8)

Reply:
> Apart from encoding, some entities are better resolved. For images and identities, we should probably display a more recognizable name than just the ID (see screen below).

Yes agreed, although I will have to make separate calls to endpoints for this since currently I extract the name of a resource from the entity url returned by the `/auth/permissions` endpoint.

Reply:
> Should we remove the resource type "Certificate" from the dropdown? It seems to be already included in the resource type "Identity".

Discussed with core team, currently we should not remove the certificate resource type. Here is an lxd [issue](https://github.com/canonical/lxd/issues/13372) for tracking purposes, the authorisation model may be simplified in the future.

Reply:
> > Apart from encoding, some entities are better resolved. For images and identities, we should probably display a more recognizable name than just the ID (see screen below).
> 
> Yes agreed, although I will have to make separate calls to endpoints for this since currently I extract the name of a resource from the entity url returned by the `/auth/permissions` endpoint.
> 
> Edit: I've made the update for the `image`, `identity` and `certificate` resource types. For those resource types, the resource select option labels are adjusted to have more readable presentation.

The same update is applied in the table below as well. The Resoruce column has content adjusted in any row with Resource Type being `image`, `identity` or `certificate`